## 1. Scrapping Usernames

In this notebook, I will be scrapping a list of usernames from MyAnimeList(MAL).  
This will be done using the Jikan unofficial MAL API.  
As there are too many users on MAL, I will only scrape the users that are a part of the largest community club.  
This club is called the "Recommendation Club" and has more than 30,000 users.  

In [1]:
import requests
import pandas as pd

In [2]:
from jikanpy import Jikan
jikan = Jikan()

In [36]:
res = requests.get('https://api.jikan.moe/v3/club/20081/')
res.json()

{'request_hash': 'request:club:b5fa34e5fe004b10fb5037abe2306786d178c8a6',
 'request_cached': False,
 'request_cache_expiry': 172800,
 'API_DEPRECATION': True,
 'API_DEPRECATION_DATE': '2022-02-10T16:02:50+00:00',
 'API_DEPRECATION_INFO': 'https://bit.ly/jikan-v3-deprecation',
 'mal_id': 20081,
 'url': 'https://myanimelist.net/clubs.php?cid=20081',
 'image_url': 'https://cdn.myanimelist.net/images/clubs/12/343236.jpg',
 'title': 'Recommendation Club',
 'members_count': 31824,
 'pictures_count': 23,
 'category': 'Anime',
 'created': '2010-03-21T00:00:00+00:00',
 'type': 'public',
 'staff': [{'mal_id': 0,
   'type': 'profile',
   'name': 'blob',
   'url': 'https://myanimelist.net/profile/blob'},
  {'mal_id': 0,
   'type': 'profile',
   'name': 'Cashdax',
   'url': 'https://myanimelist.net/profile/Cashdax'},
  {'mal_id': 0,
   'type': 'profile',
   'name': 'Confederate_Jeb',
   'url': 'https://myanimelist.net/profile/Confederate_Jeb'},
  {'mal_id': 0,
   'type': 'profile',
   'name': 'Fake

In [3]:
#at time of writing, the largest club (recommendation club) has 31,839 members or 885 pages of 36 users
# url = "https://api.jikan.moe/v3/club/1/members/1"
club_members = []
for i in range(1,886):
    res = requests.get(f"https://api.jikan.moe/v3/club/20081/members/{i}")
    data = res.json()
    for number in range(36):
        try:
            club_members.append(data['members'][number]['username'])
        except:
            break
    print(f'page {i} done and length is {len(club_members)}')
    # res.json()['members'][0]['username']
len(club_members)

page 1 done and length is 36
page 2 done and length is 72
page 3 done and length is 108
page 4 done and length is 144
page 5 done and length is 180
page 6 done and length is 216
page 7 done and length is 252
page 8 done and length is 288
page 9 done and length is 324
page 10 done and length is 360
page 11 done and length is 396
page 12 done and length is 432
page 13 done and length is 468
page 14 done and length is 504
page 15 done and length is 540
page 16 done and length is 576
page 17 done and length is 612
page 18 done and length is 648
page 19 done and length is 684
page 20 done and length is 720
page 21 done and length is 756
page 22 done and length is 792
page 23 done and length is 828
page 24 done and length is 828
page 25 done and length is 864
page 26 done and length is 900
page 27 done and length is 936
page 28 done and length is 972
page 29 done and length is 1008
page 30 done and length is 1044
page 31 done and length is 1080
page 32 done and length is 1116
page 33 done an

page 254 done and length is 8424
page 255 done and length is 8424
page 256 done and length is 8460
page 257 done and length is 8496
page 258 done and length is 8532
page 259 done and length is 8568
page 260 done and length is 8604
page 261 done and length is 8604
page 262 done and length is 8640
page 263 done and length is 8676
page 264 done and length is 8712
page 265 done and length is 8748
page 266 done and length is 8784
page 267 done and length is 8784
page 268 done and length is 8820
page 269 done and length is 8856
page 270 done and length is 8892
page 271 done and length is 8928
page 272 done and length is 8964
page 273 done and length is 9000
page 274 done and length is 9000
page 275 done and length is 9000
page 276 done and length is 9036
page 277 done and length is 9036
page 278 done and length is 9072
page 279 done and length is 9108
page 280 done and length is 9144
page 281 done and length is 9180
page 282 done and length is 9216
page 283 done and length is 9252
page 284 d

page 497 done and length is 15948
page 498 done and length is 15984
page 499 done and length is 16020
page 500 done and length is 16020
page 501 done and length is 16056
page 502 done and length is 16092
page 503 done and length is 16128
page 504 done and length is 16164
page 505 done and length is 16200
page 506 done and length is 16236
page 507 done and length is 16272
page 508 done and length is 16308
page 509 done and length is 16344
page 510 done and length is 16344
page 511 done and length is 16380
page 512 done and length is 16416
page 513 done and length is 16452
page 514 done and length is 16488
page 515 done and length is 16524
page 516 done and length is 16524
page 517 done and length is 16560
page 518 done and length is 16596
page 519 done and length is 16632
page 520 done and length is 16668
page 521 done and length is 16704
page 522 done and length is 16704
page 523 done and length is 16740
page 524 done and length is 16740
page 525 done and length is 16776
page 526 done 

page 738 done and length is 23724
page 739 done and length is 23760
page 740 done and length is 23796
page 741 done and length is 23832
page 742 done and length is 23868
page 743 done and length is 23904
page 744 done and length is 23904
page 745 done and length is 23940
page 746 done and length is 23976
page 747 done and length is 23976
page 748 done and length is 24012
page 749 done and length is 24048
page 750 done and length is 24084
page 751 done and length is 24120
page 752 done and length is 24156
page 753 done and length is 24192
page 754 done and length is 24192
page 755 done and length is 24228
page 756 done and length is 24264
page 757 done and length is 24300
page 758 done and length is 24336
page 759 done and length is 24372
page 760 done and length is 24408
page 761 done and length is 24444
page 762 done and length is 24444
page 763 done and length is 24480
page 764 done and length is 24516
page 765 done and length is 24552
page 766 done and length is 24588
page 767 done 

28384

In [4]:
club_members = pd.DataFrame(club_members)

In [5]:
club_members.head()

0
0       ---Adina---
1      ---DANIEL---
2  --Hasuki_Komai--
3        --KATAKURI
4        --Kenshin_

In [6]:
club_members.to_csv('../data/club_members.csv', index = False)

### Scrapped Data
A total of 28,384 usernames were scrapped.  
Some of the usernames were not scrapped successfully as the club has more than 30,000 users.  
This could have been due to accounts being deleted or there were issues with the request and server.  